src/contexts/NetworkFeeProvider.tsx

```diff
@@ -13,22 +15,87 @@ import { useConnectionContext } from '@src/contexts/ConnectionProvider';
 import { chainIdToCaip } from '@src/utils/caipConversion';
 
 import { useNetworkContext } from './NetworkProvider';
+import { FundTxHandler } from '@src/background/services/gasless/handlers/fundTx';
+import { GetGaslessEligibilityHandler } from '@src/background/services/gasless/handlers/getGaslessEligibility';
+import { FetchAndSolveChallengeHandler } from '@src/background/services/gasless/handlers/fetchAndSolveChallange';
+import { filter, map } from 'rxjs';
+import { gaslessChallangeUpdateEventListener } from '@src/background/services/gasless/events/gaslessChallangeUpdateListener';
+import { AddressLike, TransactionRequest } from 'ethers';
+import { useFeatureFlagContext } from './FeatureFlagsProvider';
+import { FeatureGates } from '@src/background/services/featureFlags/models';
+import { SetDefaultStateValuesHandler } from '@src/background/services/gasless/handlers/setDefaultStateValues';
+import { GaslessPhase } from '@src/background/services/gasless/model';
 
 const NetworkFeeContext = createContext<{
   networkFee: NetworkFee | null;
   getNetworkFee: (caipId: string) => Promise<NetworkFee | null>;
+  fetchAndSolveGaslessChallange: () => Promise<any | null>;
+  gaslessFundTx: ({
+    data,
+    fromAddress,
+  }: {
+    data: TransactionRequest;
+    fromAddress: string;
+  }) => Promise<string | undefined>;
+  isGaslessOn: boolean;
+  setIsGaslessOn: Dispatch<SetStateAction<boolean>>;
+  isGaslessEligible: boolean | null;
```

Comment:
The  `isGaslessEligible` and `isGaslessOn` flags can also be folded into `gaslessPhase`. Since the gasless cannot be FUNDED if its not ON and it cannot be ON if it's not ELIGIBLE.